src/services.ts

```diff
@@ -7,39 +7,41 @@ import { showSuccess, showError } from '@nextcloud/dialogs'
 import type { Node } from '@nextcloud/files'
 import { formatFileSize } from '@nextcloud/files'
 import { generateOcsUrl } from '@nextcloud/router'
-import Vue from 'vue'
+import { createApp } from 'vue'
 import Modal from './Modal.vue'
 import { loadState } from '@nextcloud/initial-state'
 import { translate as t, translatePlural as n } from '@nextcloud/l10n'
 
-Vue.prototype.t = t
-Vue.prototype.n = n
-
 const MAX_COMPRESS_SIZE = loadState('files_zip', 'max_compress_size', -1)
 
 const askForName = async (nodes: Node[]) => {
 	const modal = document.createElement('div')
 	modal.id = 'files_zip_modal'
 	document.body.appendChild(modal)
-	const App = Vue.extend(Modal)
-	const modalInstance = new App({
-		propsData: {
-			nodes,
-		},
-	})
-	modalInstance.$mount('#files_zip_modal')
+
+	let resolvePromise: (value: string | null) => void
 
 	const promise = new Promise<string|null>((resolve) => {
-		modalInstance.$on('confirm', (result: string) => {
-			modalInstance.$destroy()
-			resolve(result)
-		})
-		modalInstance.$on('closing', () => {
-			modalInstance.$destroy()
-			resolve(null)
-		})
+		resolvePromise = resolve
 	})
 
+	const app = createApp(Modal, {
+		nodes,
+		onConfirm: (result: string) => {
+			app.unmount()
+			resolvePromise(result)
+		},
+		onClosing: () => {
+			app.unmount()
+			resolvePromise(null)
+		},
+	})
+
+	app.provide('t', t)
+	app.provide('n', n)
+
```

Comment:
should not be needed anymore as you already import it directly in the component:

```suggestion
```